core/src/main/java/org/mule/runtime/core/DefaultMuleMessage.java

```diff
@@ -473,7 +474,14 @@ public int getCorrelationSequence()
     public void setCorrelationSequence(int sequence)
     {
         assertAccess(WRITE);
-        setOutboundProperty(MULE_CORRELATION_SEQUENCE_PROPERTY, sequence);
+        if (sequence >= 0)
```

Comment:
Wouldn't it be better to use nullable Integer or better Optional here? Also this assumes 0 is not a valid sequence number doesn't it?